website/src/components/quickstartGuideList/index.js

```diff
@@ -1,19 +1,68 @@
 import React from 'react';
+import { useState, useEffect, useMemo } from 'react';
 import Head from '@docusaurus/Head';
 import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
 import Layout from '@theme/Layout';
 import Hero from '@site/src/components/hero';
 import QuickstartGuideCard from '../quickstartGuideCard'
 import styles from './styles.module.css';
+import { SelectDropdown } from '../selectDropdown';
+import SearchInput from '../searchInput';
 
 const quickstartTitle = 'Quickstarts'
 const quickstartDescription = 'dbt Core is a powerful open-source tool for data transformations and dbt Cloud is the fastest and most reliable way to deploy your dbt jobs. With the help of a sample project, learn how to quickly start using dbt and one of the most common data platforms.'
 
+
 function QuickstartList({ quickstartData }) {
-  const { siteConfig } = useDocusaurusContext()
- 
+  const { siteConfig } = useDocusaurusContext();
+  const [filteredData, setFilteredData] = useState(() => quickstartData);
+  const [selectedTags, setSelectedTags] = useState([]);
+  const [selectedLevel, setSelectedLevel] = useState([]);
+  const [searchInput, setSearchInput] = useState('');
+
   // Build meta title from quickstartTitle and docusaurus config site title
-  const metaTitle = `${quickstartTitle}${siteConfig?.title ? ` | ${siteConfig.title}` : ''}`
+  const metaTitle = `${quickstartTitle}${siteConfig?.title ? ` | ${siteConfig.title}` : ''}`;
+
+  // UseMemo to prevent re-rendering on every filter change
+  // Get tag options
+  // Populated from the tags frontmatter array
+  const tagOptions = useMemo(() => {
+    const tags = new Set();
+    quickstartData.forEach(guide =>
+      guide?.data?.tags?.forEach(tag => tags.add(tag))
+    );
+    // Sort alphabetically
+    return Array.from(tags).sort((a, b) => a.toLowerCase().localeCompare(b.toLowerCase())).map(tag => ({ value: tag, label: tag }));
+  }, [quickstartData]);
```

Comment:
Would be interested in diving into this more as i'm not too familiar with `useMemo`!